tools/stress/src/org/apache/cassandra/stress/util/JavaDriverClient.java

```diff
@@ -103,22 +103,25 @@ public JavaDriverClient(StressSettings settings, List<String> hosts, int port, E
     private LoadBalancingPolicy loadBalancingPolicy(StressSettings settings)
     {
         LoadBalancingPolicy ret = null;
+        ReplicaOrdering replicaOrdering = null;
 
         if (settings.node.rack != null) {
             RackAwareRoundRobinPolicy.Builder policyBuilder = RackAwareRoundRobinPolicy.builder();
             if (settings.node.datacenter != null)
                 policyBuilder.withLocalDc(settings.node.datacenter);
             policyBuilder = policyBuilder.withLocalRack(settings.node.rack);
             ret = policyBuilder.build();
+            replicaOrdering = ReplicaOrdering.NEUTRAL;
```

Comment:
so this mean that a mix of using Rackaware and tablets would be imbalanced ? 

and would be needed to be fix on the driver end ?

Reply:
All round robin policies (rack,dc) used with TokenAwarePolicy can be imbalanced with tablets when using neutral ordering. I think this combination should not be used if we want load to be as balanced as possible. Long story short let's say we have RF=3 and 6 nodes [A,B,C,D,E,F] and tablets are spread evenly but only on ABC (this can happen when growing the cluster). If round robin happens to point to either D,E,F,A then that request will hit replica A first. This results in A getting 4/6 of the load, B 1/6, and C 1/6. However if RF=3 and cluster has only A,B,C then all will be perfectly balanced.

I'll try to make a comment with broader explanation what happens in https://github.com/scylladb/scylladb/issues/19107 to better illustrate this issue with neutral ordering.

The rack aware one just does not work correctly (will ignore rack awareness in favor of local dc) with random ordering so it has to stay neutral for now.